### PR TITLE
Kkraune/ranking

### PR DESCRIPTION
**Pull request by kkraune**
rewrote for better flow, added some links

### CODE REVIEW

File: documentation/ranking.html [hunk 1]
```diff
@@ -45,15 +47,20 @@
...
-this can be used for implementing different use cases, or bucket testing ranking variations.
+If not specified, the <a href="text-ranking.html">default text ranking</a> profile is used.
 </p><p>
 A rank profile can <em>inherit</em> another rank profile.
 </p><p>
-If rank-profile is not specified, the <a href="text-ranking.html">default text ranking</a> is used.
+Queries select profile using <a href="reference/search-api-reference.html#ranking.profile">ranking.profile</a>,
+or in <a href="searcher-development.html">Searcher</a> code:
+<pre>
+query.properties().set(new CompoundName("ranking.profile"), "my_rank-profile");
```
Comment (jobergum) [inline documentation/ranking.html]: There is a direct programatic api for this, query.getRanking().setProfile("my-rank-profile"), must check syntax.

File: documentation/ranking.html [hunk 2]
```diff
@@ -5,14 +5,16 @@
 
 <p>
 Vespa uses <a href="reference/ranking-expressions.html">ranking expression</a>
-to rank documents matching a query, computing a rank score per document.
-The results of the ranking computation can be returned with the documents,
-and used to order and further select the documents that should be returned.
-Simple example: Rank the freshest document highest, using the value in an attribute named <em>timestamp</em>:
+to rank documents matching a query, computing a rank score per document using <a href="#rank-features">rank features</a>.
```
Comment (jobergum) [inline documentation/ranking.html]: using ranking expressions, not rank features, but an expression will likely use rank-features.

File: documentation/ranking.html [hunk 3]
```diff
@@ -45,15 +47,20 @@
...
 A rank profile can <em>inherit</em> another rank profile.
 </p><p>
-If rank-profile is not specified, the <a href="text-ranking.html">default text ranking</a> is used.
+Queries select profile using <a href="reference/search-api-reference.html#ranking.profile">ranking.profile</a>,
+or in <a href="searcher-development.html">Searcher</a> code:
+<pre>
+query.properties().set(new CompoundName("ranking.profile"), "my_rank-profile");
+</pre>
+Note that some use cases performs better using the
+<a href="reference/search-definitions-reference.html#rank-profile">unranked</a> rank profile.
```
Comment (jobergum) [inline documentation/ranking.html]: But then your hits are unranked, random order unless sorting is specified.

File: documentation/ranking.html (hunk 1 above)
Comment (kkraune) [reply documentation/ranking.html]: yup, query.getRanking().setProfile("my_rank-profile"); works - thanks!

File: documentation/ranking.html [hunk 4]
```diff
@@ -57,7 +57,7 @@ <h2 id="rank-profile">Rank profile</h2>
 Queries select profile using <a href="reference/search-api-reference.html#ranking.profile">ranking.profile</a>,
 or in <a href="searcher-development.html">Searcher</a> code:
 <pre>
-query.properties().set(new CompoundName("ranking.profile"), "my_rank-profile");
+query.getRanking().setProfile("my_rank-profile");
```
Comment (jobergum) [inline documentation/ranking.html]: I think it should my-rank-profile

**Pull request merged**